crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -43,6 +43,36 @@ impl SystemMeta {
         &self.name
     }
 
+    /// Returns the system's component access set.
+    #[inline]
+    pub fn component_access_set(&self) -> &FilteredAccessSet<ComponentId> {
+        &self.component_access_set
+    }
+
+    /// Returns a mutable reference to the system's component access set.
+    ///
+    /// # Safety
+    /// This allows unsafe modifications to the component access set that can violate Bevy's scheduling soundness.
```

Comment:
This isn't documenting the actual safety invariant. Editing the component access set needs to accurately reflect the actual mutable and immutable accesses the system (or parameter) fetches from the World.